Initial HTTP POST/PUT support

renshao opened the pull request:
- Default all POST/PUT request with 'Content-Type: application/json'
  header
- Each POST/PUT request takes one line from payloads file

markpritchard commented on src/config.rs:
```diff
@@ -97,6 +122,21 @@ impl Config {
+                .short('m')
+                .long("method")
+                .value_name("http_method")
+                .possible_values(&["GET", "POST", "PUT", "DELETE"])
```
Lets drop `DELETE` from this ... I know its for completeness, but

renshao replied:
@markpritchard 
I've removed `DELETE`. Also addressed a few clippy suggestions

Commit:
Initial HTTP POST/PUT support

- Default all POST/PUT request with 'Content-Type: application/json'
  header
- Each POST/PUT request takes one line from payloads file
- Addressed some of suggestions in clippy

markpritchard commented:
Merged in #11
